src/wasm/wasm.cpp

```diff
@@ -1441,8 +1441,12 @@ void StackSwitch::finalize() {
   }
 
   assert(this->cont->type.isContinuation());
-  type =
+  Type params =
     this->cont->type.getHeapType().getContinuation().type.getSignature().params;
+  assert(params.size() > 0);
+  Type cont = params[params.size() - 1];
+  assert(cont.isContinuation());
+  type = cont.getHeapType().getContinuation().type.getSignature().params;
```

Comment:
@dhil Does this look correct to you?

Reply:
Yes, it looks right to me. The last parameter of a switch-continuation must be a continuation type. This argument is synthetic in the sense that it is provided by `switch` rather than the programmer.